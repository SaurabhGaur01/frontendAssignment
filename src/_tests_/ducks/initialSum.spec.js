import {
    setInitialSum,
    initialSumReducer,
} from '../../ducks/initialSum';

const mockKey = 'mockValue';
 
describe('initialSum Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setInitialSum should match the snapshot' , () => {
            expect(setInitialSum(mockKey)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(initialSumReducer()).toEqual(10000);
        });
        it('should return the passed state if called with no action' , () => {
            expect(initialSumReducer(mockKey)).toEqual(mockKey);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(initialSumReducer(mockKey, unknownAction)).toEqual(mockKey);
        });
        it('should return the value called with' , () => {
            expect(initialSumReducer('', setInitialSum(mockKey))).toEqual(mockKey);
        });
    });
});