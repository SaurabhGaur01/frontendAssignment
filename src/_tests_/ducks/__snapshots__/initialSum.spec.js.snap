// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initialSum Duck Tests Action Generator setInitialSum should match the snapshot 1`] = `
Object {
  "type": "modules/app/SET_INITIAL_SUM",
  "value": "mockValue",
}
`;
